src/pydvl/value/shapley/gt.py

```diff
@@ -277,56 +281,66 @@ def reducer(
 
     ###########################################################################
     # Solution of the constraint problem with scipy
+    # DISABLED: highs fails to find solutions that SCS finds
+
+    # A_ub, b_ub = _build_gt_constraints(n, bound=epsilon / (2 * np.sqrt(n)), C=C)
+    # c = np.zeros_like(u.data.indices)
+    # # A trivial bound for the values from the definition is max_utility * (n-1)
+    # bounds = tuple(u.score_range * n)  # u.score_range defaults to (-inf, inf)
+    # result: sp.optimize.OptimizeResult = sp.optimize.linprog(
+    #     c,
+    #     A_ub=A_ub,
+    #     b_ub=b_ub,
+    #     A_eq=np.ones((1, n)),
+    #     b_eq=total_utility,
+    #     bounds=bounds,
+    #     method="highs",
+    #     options={},
+    # )
 
-    A_ub, b_ub = _build_gt_constraints(n, bound=eps / (2 * np.sqrt(n)), C=C)
-    c = np.zeros_like(u.data.indices)
-    total_utility = u(u.data.indices)
-    # A trivial bound for the values from the definition is max_utility * (n-1)
-    bounds = tuple(u.score_range * n)  # u.score_range defaults to (-inf, inf)
-    result: sp.optimize.OptimizeResult = sp.optimize.linprog(
-        c,
-        A_ub=A_ub,
-        b_ub=b_ub,
-        A_eq=np.ones((1, n)),
-        b_eq=total_utility,
-        bounds=bounds,
-        method="highs",
-        options={},
+    ###########################################################################
+    # Solution of the constraint problem with cvxpy
+
+    import cvxpy as cp
+
+    v = cp.Variable(n)
+    constraints = [cp.sum(v) == total_utility]
+    for i in range(n):
+        for j in range(i + 1, n):
+            constraints.append(v[i] - v[j] <= epsilon + C[i, j])
+            constraints.append(v[j] - v[i] <= epsilon - C[i, j])
+
+    problem = cp.Problem(cp.Minimize(0), constraints)
+    problem.solve(solver=cp.SCS)
```

Comment:
We should allow the user to change just like in the least core methods.

Reply:
On it